### PR TITLE
[ADD] sale_order_revision_multi: New module

### PR DESCRIPTION
**Pull request by leemannd**
This module is based upon `sale_order_revision`.
- this module allows to create several revisions of a quotation as `sale_order_revision`
- all these revisions will be active (unless the OCA module)
- once a revision is validated of them is validated, it will cancel all other linked revisions

### CODE REVIEW

**Review by cyrilmanuel — APPROVED** [review]

**Comment by cyrilmanuel** [comment]
not an expert but LGTM

**Review by Ricardoalso — COMMENTED** [review]
missing import statement in sale_order_revision_multi/tests/__init__.py

**Review by Ricardoalso — APPROVED** [review]
Comment above about missing import, otherwise LGTM

File: sale_order_revision_multi/models/sale_order.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+            "sale.act_res_partner_2_sale_order"
+        )
+        action["domain"] = [
+            ("unrevisioned_name", "=", self.unrevisioned_name),
+            ("id", "!=", self.id),
+        ]
+        return action
+
+    so_variant_ids = fields.One2many(
+        "sale.order", compute="_compute_so_variant_ids", string="Variants"
```
Comment (simahawk) [inline sale_order_revision_multi/models/sale_order.py:31]: "variant" is actually a "revision"?
If yes, I would rename all strings :wink:

File: sale_order_revision_multi/models/sale_order.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
+# Copyright 2022 Camptocamp SA (<https://www.camptocamp.com>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import fields, models
+
+
+class SaleOrder(models.Model):
+    _inherit = ["sale.order"]
+
+    def _compute_so_variant_ids(self):
```
Comment (simahawk) [inline sale_order_revision_multi/models/sale_order.py:10]: pls move methods below fields

File: sale_order_revision_multi/models/sale_order.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        for order in self:
+            variant_orders = self.search(
```
Comment (simahawk) [inline sale_order_revision_multi/models/sale_order.py:12]: for counters is always better to use read_group

File: sale_order_revision_multi/views/sale_order_view.xml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+      <xpath expr="//field[@name='current_revision_id']" position="attributes">
+        <attribute name="invisible">1</attribute>
+      </xpath>
+
+      <!-- override the create revision button to make is visible when it's a quotation or when it is canceled -->
+      <xpath expr="//button[@name='create_revision']" position="attributes">
+        <attribute name="states">draft,sent,cancel</attribute>
+      </xpath>
+
+      <!-- override the always hide this button -->
```
Comment (simahawk) [inline sale_order_revision_multi/views/sale_order_view.xml:30]:
```suggestion
      <!-- override to always hide this button -->
```

File: sale_order_revision_multi/tests/test_sale_order_revision.py [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+    def _create_tester(self):
+        sale_form = Form(self.revision_model)
+        sale_form.partner_id = self.partner
+        with sale_form.order_line.new() as line_form:
+            line_form.product_id = self.product
+        return sale_form.save()
+
+    def test_revision_preserve_state(self):
+        """Creating a new revision should not change the state of the SO from which it
+        is generated
```
Comment (simahawk) [inline sale_order_revision_multi/tests/test_sale_order_revision.py:25]:
```suggestion
        """New revision create does not change state of original SO.
```

File: sale_order_revision_multi/tests/test_sale_order_revision.py [hunk 6]
```diff
@@ -0,0 +1,126 @@
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
```
Comment (simahawk) [inline sale_order_revision_multi/tests/test_sale_order_revision.py:1]: copyright missing

File: sale_order_revision_multi/__manifest__.py [hunk 7]
```diff
@@ -0,0 +1,14 @@
+{
+    "name": "Sale order revisions multi",
+    "summary": "Keep track of several active revised quotations",
```
Comment (simahawk) [inline sale_order_revision_multi/__manifest__.py:3]:
```suggestion
    "summary": "Manage multiple active revisions for quotations",
```

**Review by simahawk — CHANGES_REQUESTED** [review]

**Comment by dreispt** [comment]
This sounds more like "options" than "revisions".

**Comment by leemannd** [comment]
It's maybe a better way to define them.
This is maybe not the best way of dealing with it.

**Comment by leemannd** [comment]
Closing the PR. Feel free to re-open it

**Pull request closed without merging**